packages/components/src/button/index.tsx

```diff
@@ -78,7 +81,7 @@ const Button: ForwardRefRenderFunction<
 				'is-borderless': props.borderless,
 		  } );
 
-	if ( isAnchor( props ) ) {
+	if ( isAnchor( props ) || props.renderAs === 'a' ) {
```

Comment:
Maybe I'm missing something but this way it wouldn't fix the issue for code like this:
` <Button renderAs="button" href={url} />` . Maybe it should be `isAnchor( props ) && props.renderAs === 'a'`

Reply:
`isAnchor( props ) && props.renderAs === 'a'` will fail for existing consumers of the component.

Reply:
Yes, so we would default it to `'a'` . Because `if ( isAnchor( props ) || props.renderAs === 'a' ) ` won't allow us to skip the anchor logic if we would want to.

Reply:
> Because `if ( isAnchor( props ) || props.renderAs === 'a' ) ` won't allow us to skip the anchor logic if we would want to.

If the consumer doesn't pass `renderAs`, it will fallback to the previous behavior.